parquet-variant/src/from_json.rs

```diff
@@ -85,19 +85,19 @@ fn variant_from_number<'a, 'b>(n: &Number) -> Result<Variant<'a, 'b>, ArrowError
 
 fn append_json(json: &Value, builder: &mut impl AppendVariantHelper) -> Result<(), ArrowError> {
     match json {
-        Value::Null => builder.append_value_helper(Variant::Null),
-        Value::Bool(b) => builder.append_value_helper(*b),
+        Value::Null => builder.append_value(Variant::Null),
+        Value::Bool(b) => builder.append_value(*b),
         Value::Number(n) => {
-            builder.append_value_helper(variant_from_number(n)?);
+            builder.append_value(variant_from_number(n)?);
         }
-        Value::String(s) => builder.append_value_helper(s.as_str()),
+        Value::String(s) => builder.append_value(s.as_str()),
         Value::Array(arr) => {
-            let mut list_builder = builder.new_list_helper();
+            let mut list_builder = builder.new_list();
             build_list(arr, &mut list_builder)?;
```

Comment:
Re https://github.com/apache/arrow-rs/pull/7783#discussion_r2168925972, this is what I meant by "fold the function into its call site":
```suggestion
            for val in arr {
                append_json(val, &builder)?;
            }
```

Reply:
Yeah, already done